compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -2642,6 +2642,7 @@ boolean checkTypeAndVarCountConsistency(BLangTupleVariable varNode, BTupleType t
             dlog.error(varNode.pos, DiagnosticErrorCode.NO_NEW_VARIABLES_VAR_ASSIGNMENT);
             return false;
         }
+        varNode.setBType(tupleTypeNode);
```

Comment:
Why is this fixed only for list binding patterns? Need to fix for mapping binding patterns too.

I believe this is handled already for error binding patterns, but let's add a simple test for that too.

Reply:
Came across a separate issue for error binding patterns which needs to be fixed separately - https://github.com/ballerina-platform/ballerina-lang/issues/40178.